### PR TITLE
[All] - unit tests appear as ok even though non zero exit status code

### LINKED ISSUE #3071: [ALL] : Unit tests appear as OK even though non-zero exit status code
## Description

Running https://github.com/OpenCTI-Platform/connectors/blob/eb52b02912091976c5e465ac7e64dca18ea1deab/run_test.sh always returns zero-exit code.

## Environment

1. OS (where OpenCTI server runs): NA
2. OpenCTI version: NA
3. OpenCTI client: NA
4. Other environment details:

## Reproducible Steps

Steps to create the smallest reproducible scenario:
1. Add a failing test (e.g. `assert False`) in any test_xxx.py
2. Run bash run_test.sh
3. Observe exit code 0

## Expected Output

A non zero exit code if a test (or the dependency resolution) failed

## Actual Output

Always a zero exit code

## Additional information

Same happens in CircleCI run

## Screenshots (optional)
Image

### PR DESCRIPTION
**Pull request by flavienSindou**
### Proposed changes

* Create isolated environments for monorepo subprojects testing
* exit on non zero exit
*  Remove unused test-requirements and dummy tests
* Revert Bot Renovate bad updates.

### Related issues

* https://github.com/OpenCTI-Platform/connectors/issues/3071

### Checklist

- [x] I consider the submitted work as finished
- [x] I tested the code for its functionality using different use cases
- [ ] I added/update the relevant documentation (either on github or on notion)
- [x] Where necessary I refactored code to improve the overall quality

### Further comments

Note : we ensure this can be run via the CI but also locally, with different OS.

Compared to what is reported in the issue description, we now detect dependency resolution errors and pytest non zero exit code

image

image

### CODE REVIEW

File: run_test.sh
Comment (flavienSindou) [inline run_test.sh:1]: Core of the PR

File: run_test.sh [hunk 1]
```diff
@@ -1,18 +1,42 @@
...
+  project=$(echo "$requirements_file" | cut -d'/' -f1-3)
+  echo 'Running tests pipeline for project' $project
+
+  echo 'Creating isolated virtual environment'
+  python -m venv "$venv_name"
+  if [ -f "$venv_name/bin/activate" ]; then
+    source "$venv_name/bin/activate"  # Linux/MacOS
+  elif [ -f "$venv_name/Scripts/activate" ]; then
+    source "$venv_name/Scripts/activate"  # Windows
+  fi
```
Comment (flavienSindou) [inline run_test.sh:28]: Ensure it can be run locally by windows users

File: stream/crowdstrike-endpoint-security/src/requirements.txt [hunk 2]
```diff
@@ -1,3 +1,3 @@
 pycti==6.4.2
-prometheus-client==0.21.1
+prometheus-client~=0.20.0
```
Comment (flavienSindou) [inline stream/crowdstrike-endpoint-security/src/requirements.txt:2]: fix dependency resolution issue (renovate bot update to 0.21 bu pycti only compatible with 0.20)

File: run_test.sh (hunk 1 above)
Comment (Powlinett) [reply run_test.sh:28]: Tested both on Windows and Ubuntu (WSL), works as intended 👍

File: run_test.sh
Comment (Powlinett) [reply run_test.sh:1]: LGTM 👍

**Review by Powlinett — APPROVED** [review]

**Review by Megafredo — APPROVED** [review]
It's okay with me too ! Great job !

**Pull request merged**